cnf-certification-test/accesscontrol/suite.go

```diff
@@ -410,44 +410,58 @@ func testPodRoleBindings(env *provider.TestEnvironment) {
 			continue
 		}
 
-		// Get any rolebindings that do not belong to the pod namespace.
-		roleBindings, err := rbac.GetRoleBindings(put.Namespace, put.Spec.ServiceAccountName)
-		if err != nil {
-			failedPods = append(failedPods, put.Name)
-		}
+		logrus.Infof("%s has a serviceAccountName: %s, checking role bindings.", put.String(), put.Spec.ServiceAccountName)
+
+		// Loop through the rolebindings and check if they are from another namespace
+		for rbIndex := range env.RoleBindings {
+			found := false
+			for _, subject := range env.RoleBindings[rbIndex].Subjects {
+				// Match the subject service account with the name of the pod's service account.
+				// Determine if that service account is not in the same namespace as the pod.
+				if subject.Kind != "ServiceAccount" || subject.Name != put.Spec.ServiceAccountName {
+					continue
+				}
 
-		if len(roleBindings) > 0 {
-			logrus.Warnf("Pod: %s/%s has the following role bindings: %s", put.Namespace, put.Name, roleBindings)
-			tnf.ClaimFilePrintf("Pod: %s/%s has the following role bindings: %s", put.Namespace, put.Name, roleBindings)
-			failedPods = append(failedPods, put.Name)
+				if rbac.IsRoleBindingOutOfNamespace(put.Namespace, put.Spec.ServiceAccountName, env.RoleBindings[rbIndex].Namespace, env.RoleBindings[rbIndex].Name) {
+					logrus.Warnf("Pod: %s/%s has the following role bindings that do not live in the same namespace: %s", put.Namespace, put.Name, env.RoleBindings[rbIndex].Name)
+					tnf.ClaimFilePrintf("Pod: %s/%s has the following role bindings that do not live in the same namespace: %s", put.Namespace, put.Name, env.RoleBindings)
+					failedPods = append(failedPods, put.Name)
+					found = true
+					break
+				}
+			}
+			// Break of out the loop if we found a role binding that is out of namespace
+			if found {
+				break
+			}
 		}
 	}
 	testhelper.AddTestResultLog("Non-compliant", failedPods, tnf.ClaimFilePrintf, ginkgo.Fail)
```

Comment:
use the new reason object, see testContainerSCC function.
https://github.com/test-network-function/cnf-certification-test/pull/834

Reply:
Done.

Reply:
@edcdavid Can you re-review to see if I'm using it the way it is intended?  Thanks!